{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "## **Downloading the data from online**"
      ],
      "metadata": {
        "id": "i-YnX4W3iQlo"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PWsXZg5zBuZR",
        "outputId": "26af5d58-b9ac-4cb7-cfb9-0cbc54e030cc"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1-19R0FnpDSljtWkNKCKiciDW8HESq03P\n",
            "From (redirected): https://drive.google.com/uc?id=1-19R0FnpDSljtWkNKCKiciDW8HESq03P&confirm=t&uuid=861bf890-ac04-41a0-8f84-99fbcc88052e\n",
            "To: /content/HumanWGS.SUBSET.R2.fastq\n",
            "100% 695M/695M [00:04<00:00, 140MB/s]\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=10n-w54s2d32aAGJKrgKjm5d8EMVpy-45\n",
            "From (redirected): https://drive.google.com/uc?id=10n-w54s2d32aAGJKrgKjm5d8EMVpy-45&confirm=t&uuid=38887140-85f2-4f0e-b4f7-80ca33bd9ee6\n",
            "To: /content/HumanWGS.SUBSET.R1.fastq\n",
            "100% 695M/695M [00:04<00:00, 155MB/s]\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1-0aDSqXLDsrNCEGPRTO4ppqx-0HF-f8R\n",
            "From (redirected): https://drive.google.com/uc?id=1-0aDSqXLDsrNCEGPRTO4ppqx-0HF-f8R&confirm=t&uuid=db90ac5d-cb80-4e45-bcf9-3d6454c47e25\n",
            "To: /content/Human38.tar.gz\n",
            "100% 4.35G/4.35G [00:39<00:00, 109MB/s]\n"
          ]
        }
      ],
      "source": [
        "#### Human Sample\n",
        "#### This dataset has a small error, the names for both read ends do not match, it was downloaded from NCBI and a fastq interleaved, which means that reads are all in a single file, so\n",
        "#### but I split in two files, which is the most common format, but didnt notice the read ends didn't match.  Please use the dataset below\n",
        "\n",
        "#DO NOT USE #!gdown 1-QHpki7f9MjNbMMQoMjojfKi8OJ9BwrQ\n",
        "#DO NOT USE #!gdown 1-M1ogw7lNcF92Uldl6I4rSu89q_24Y7Q\n",
        "\n",
        "##Use this, since it is fairly late, the dataset is very small, 1/6 the original size, so it will runa all alot faster.\n",
        "\n",
        "\n",
        "#reads\n",
        "!gdown  1-19R0FnpDSljtWkNKCKiciDW8HESq03P\n",
        "!gdown  10n-w54s2d32aAGJKrgKjm5d8EMVpy-45\n",
        "\n",
        "### To make thinngs faster for the student, here is the indexed human genome, hg38.\n",
        "### You have to donwload it and decompressed it, you will get both the fasta and the BWT'ed files. Note all have same prefix Human<.files>.\n",
        "### Next semester we will cover how specialized computers are often used to solve this aligmnet problem faster.\n",
        "\n",
        "\n",
        "#reference genome\n",
        "!gdown 1-0aDSqXLDsrNCEGPRTO4ppqx-0HF-f8R"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!ls"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JzeisJ2-CH5j",
        "outputId": "3035f333-d95e-443b-c590-8b3cad6a4638"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Human38.tar.gz\tHumanWGS.SUBSET.R1.fastq  HumanWGS.SUBSET.R2.fastq  sample_data\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#new programs!\n",
        "!apt install bwa\n",
        "!apt install samtools\n",
        "!apt install bcftools"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5VqSZsYSCJLV",
        "outputId": "a2391bdc-faee-4b31-94a4-7d9978743c74"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Suggested packages:\n",
            "  samtools\n",
            "The following NEW packages will be installed:\n",
            "  bwa\n",
            "0 upgraded, 1 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 195 kB of archives.\n",
            "After this operation, 466 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bwa amd64 0.7.17-6 [195 kB]\n",
            "Fetched 195 kB in 0s (801 kB/s)\n",
            "Selecting previously unselected package bwa.\n",
            "(Reading database ... 121752 files and directories currently installed.)\n",
            "Preparing to unpack .../bwa_0.7.17-6_amd64.deb ...\n",
            "Unpacking bwa (0.7.17-6) ...\n",
            "Setting up bwa (0.7.17-6) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libhts3 libhtscodecs2\n",
            "Suggested packages:\n",
            "  cwltool\n",
            "The following NEW packages will be installed:\n",
            "  libhts3 libhtscodecs2 samtools\n",
            "0 upgraded, 3 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 963 kB of archives.\n",
            "After this operation, 2,270 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtscodecs2 amd64 1.1.1-3 [53.2 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]\n",
            "Fetched 963 kB in 0s (4,252 kB/s)\n",
            "Selecting previously unselected package libhtscodecs2:amd64.\n",
            "(Reading database ... 121769 files and directories currently installed.)\n",
            "Preparing to unpack .../libhtscodecs2_1.1.1-3_amd64.deb ...\n",
            "Unpacking libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../libhts3_1.13+ds-2build1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../samtools_1.13-4_amd64.deb ...\n",
            "Unpacking samtools (1.13-4) ...\n",
            "Setting up libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Setting up libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Setting up samtools (1.13-4) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Suggested packages:\n",
            "  python3-numpy python3-matplotlib texlive-latex-recommended\n",
            "The following NEW packages will be installed:\n",
            "  bcftools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 697 kB of archives.\n",
            "After this operation, 2,360 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bcftools amd64 1.13-1 [697 kB]\n",
            "Fetched 697 kB in 0s (3,587 kB/s)\n",
            "Selecting previously unselected package bcftools.\n",
            "(Reading database ... 121868 files and directories currently installed.)\n",
            "Preparing to unpack .../bcftools_1.13-1_amd64.deb ...\n",
            "Unpacking bcftools (1.13-1) ...\n",
            "Setting up bcftools (1.13-1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# !gzip -df HumanWGS.SUBSET.R2.fastq\n",
        "# !gzip -df HumanWGS.SUBSET.R1.fastq\n",
        "!gzip -df Human38.tar.gz"
      ],
      "metadata": {
        "id": "RizfTR0lCSmy"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!tar -xf Human38.tar"
      ],
      "metadata": {
        "id": "OthJ2Y2LDA8N"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bwa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "djKh6iVBDNrM",
        "outputId": "1a0056c0-4f9d-4472-b270-48cf53d9d92c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Program: bwa (alignment via Burrows-Wheeler transformation)\n",
            "Version: 0.7.17-r1188\n",
            "Contact: Heng Li <lh3@sanger.ac.uk>\n",
            "\n",
            "Usage:   bwa <command> [options]\n",
            "\n",
            "Command: index         index sequences in the FASTA format\n",
            "         mem           BWA-MEM algorithm\n",
            "         fastmap       identify super-maximal exact matches\n",
            "         pemerge       merge overlapping paired ends (EXPERIMENTAL)\n",
            "         aln           gapped/ungapped alignment\n",
            "         samse         generate alignment (single ended)\n",
            "         sampe         generate alignment (paired ended)\n",
            "         bwasw         BWA-SW for long queries\n",
            "\n",
            "         shm           manage indices in shared memory\n",
            "         fa2pac        convert FASTA to PAC format\n",
            "         pac2bwt       generate BWT from PAC\n",
            "         pac2bwtgen    alternative algorithm for generating BWT\n",
            "         bwtupdate     update .bwt to the new format\n",
            "         bwt2sa        generate SA from BWT and Occ\n",
            "\n",
            "Note: To use BWA, you need to first index the genome with `bwa index'.\n",
            "      There are three alignment algorithms in BWA: `mem', `bwasw', and\n",
            "      `aln/samse/sampe'. If you are not sure which to use, try `bwa mem'\n",
            "      first. Please `man ./bwa.1' for the manual.\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -lh"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q9A4uX6uDaFj",
        "outputId": "f5fe7d6c-a656-4763-f442-919a5063f923"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 9.3G\n",
            "drwx------ 6 root root 4.0K Apr 12 20:39 drive\n",
            "drwxr-xr-x 2 root root 4.0K Nov 19 03:38 Human38\n",
            "-rw-r--r-- 1 root root 8.0G Apr 12 20:41 Human38.tar\n",
            "-rw-r--r-- 1 root root 663M Apr 12 20:40 HumanWGS.SUBSET.R1.fastq\n",
            "-rw-r--r-- 1 root root 663M Apr 12 20:40 HumanWGS.SUBSET.R2.fastq\n",
            "drwxr-xr-x 1 root root 4.0K Apr 11 13:21 sample_data\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## **Aligning the ends of the reads**"
      ],
      "metadata": {
        "id": "FjoUaZO-iI0N"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!bwa mem Human38/Human HumanWGS.SUBSET.R2.fastq > fwd.sai\n",
        "\n",
        "#alignment on one end of the fastq file"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZQDZQpjRDc-m",
        "outputId": "9a7ad638-4ebb-4d8a-d1a9-c8141e294a3b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 64.951 CPU sec, 67.922 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 60.206 CPU sec, 61.182 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 54.864 CPU sec, 55.751 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.576 CPU sec, 43.360 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 45.320 CPU sec, 46.004 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 46.678 CPU sec, 47.705 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.534 CPU sec, 43.267 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 43.774 CPU sec, 44.796 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 43.730 CPU sec, 44.503 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 47.096 CPU sec, 47.838 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 67.182 CPU sec, 68.396 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 54.701 CPU sec, 56.736 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 48.026 CPU sec, 49.869 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 54.916 CPU sec, 56.074 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 45.198 CPU sec, 46.159 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 51.218 CPU sec, 51.965 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 46.923 CPU sec, 47.725 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.617 CPU sec, 43.590 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 45.116 CPU sec, 45.965 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 62.367 CPU sec, 63.638 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 47.847 CPU sec, 48.806 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 64.390 CPU sec, 65.449 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 58.382 CPU sec, 59.420 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 171.038 CPU sec, 176.390 real sec\n",
            "[M::process] read 24750 sequences (2499750 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 135.597 CPU sec, 138.275 real sec\n",
            "[M::mem_process_seqs] Processed 24750 reads in 9.671 CPU sec, 9.713 real sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa mem Human38/Human HumanWGS.SUBSET.R2.fastq\n",
            "[main] Real time: 1566.382 sec; CPU: 1504.508 sec\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bwa mem Human38/Human HumanWGS.SUBSET.R1.fastq > rev.sai\n",
        "\n",
        "#alignment on the other end of the fastq file"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Bkn4n-2uDh3H",
        "outputId": "88716e8d-21ad-40bf-c431-21a6a19a7642"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 55.271 CPU sec, 56.929 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 57.123 CPU sec, 58.025 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 53.367 CPU sec, 54.456 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 39.417 CPU sec, 40.089 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 39.925 CPU sec, 40.926 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.532 CPU sec, 43.355 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.701 CPU sec, 43.409 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 40.076 CPU sec, 40.878 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 45.900 CPU sec, 46.917 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 46.117 CPU sec, 46.908 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 62.947 CPU sec, 64.086 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 52.209 CPU sec, 52.776 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 44.019 CPU sec, 44.801 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 53.398 CPU sec, 54.403 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 40.121 CPU sec, 40.911 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 49.661 CPU sec, 50.761 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 43.685 CPU sec, 44.431 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.275 CPU sec, 43.377 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 46.007 CPU sec, 47.711 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 61.752 CPU sec, 63.731 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 51.903 CPU sec, 54.715 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 57.250 CPU sec, 58.781 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 52.069 CPU sec, 53.755 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 146.935 CPU sec, 150.714 real sec\n",
            "[M::process] read 24750 sequences (2499750 bp)...\n",
            "[M::mem_process_seqs] Processed 99010 reads in 122.846 CPU sec, 125.862 real sec\n",
            "[M::mem_process_seqs] Processed 24750 reads in 10.278 CPU sec, 11.197 real sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa mem Human38/Human HumanWGS.SUBSET.R1.fastq\n",
            "[main] Real time: 1474.770 sec; CPU: 1409.401 sec\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bwa mem Human38/Human HumanWGS.SUBSET.R1.fastq HumanWGS.SUBSET.R2.fastq > Human.sam\n",
        "\n",
        "#combining both ends of the file into a sam file"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1k0VGY0oDih3",
        "outputId": "2614243e-c3ab-485d-e7c0-1e21ea184102"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 41818, 0, 3)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 400, 415)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 695)\n",
            "[M::mem_pestat] mean and std.dev: (346.62, 106.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 835)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 52.985 CPU sec, 55.377 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 39153, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 706)\n",
            "[M::mem_pestat] mean and std.dev: (348.41, 108.82)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 850)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 100.424 CPU sec, 102.653 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 34698, 1, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (276, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 699)\n",
            "[M::mem_pestat] mean and std.dev: (348.47, 107.72)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 840)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 104.929 CPU sec, 106.594 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43080, 1, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (278, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 695)\n",
            "[M::mem_pestat] mean and std.dev: (348.75, 106.77)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 834)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 43.372 CPU sec, 44.342 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 44233, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (277, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 697)\n",
            "[M::mem_pestat] mean and std.dev: (348.63, 107.58)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 837)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.811 CPU sec, 43.321 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34417, 0, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 703)\n",
            "[M::mem_pestat] mean and std.dev: (347.62, 107.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 846)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 84.905 CPU sec, 86.274 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 44327, 0, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (278, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 698)\n",
            "[M::mem_pestat] mean and std.dev: (349.54, 107.83)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 838)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.901 CPU sec, 43.472 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 43897, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 701)\n",
            "[M::mem_pestat] mean and std.dev: (348.08, 108.04)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 843)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.922 CPU sec, 43.857 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 43788, 0, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 701)\n",
            "[M::mem_pestat] mean and std.dev: (347.93, 107.75)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 843)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 44.109 CPU sec, 44.645 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43146, 0, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (278, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 698)\n",
            "[M::mem_pestat] mean and std.dev: (349.33, 107.69)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 838)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 45.459 CPU sec, 46.110 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 43352, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 706)\n",
            "[M::mem_pestat] mean and std.dev: (348.47, 108.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 850)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 47.912 CPU sec, 50.106 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41232, 2, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 703)\n",
            "[M::mem_pestat] mean and std.dev: (347.85, 108.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 846)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 53.511 CPU sec, 54.357 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 43109, 0, 2)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 706)\n",
            "[M::mem_pestat] mean and std.dev: (348.68, 108.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 850)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 47.185 CPU sec, 47.896 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 43883, 1, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 706)\n",
            "[M::mem_pestat] mean and std.dev: (348.72, 108.86)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 850)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.973 CPU sec, 43.674 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41511, 1, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (278, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 698)\n",
            "[M::mem_pestat] mean and std.dev: (348.74, 107.85)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 838)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 46.075 CPU sec, 46.590 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 42440, 0, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (281, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (7, 692)\n",
            "[M::mem_pestat] mean and std.dev: (350.23, 107.61)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 829)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.615 CPU sec, 43.433 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43442, 2, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 701)\n",
            "[M::mem_pestat] mean and std.dev: (347.79, 108.03)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 843)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 43.963 CPU sec, 44.543 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41726, 2, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 401, 416)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 700)\n",
            "[M::mem_pestat] mean and std.dev: (347.30, 107.97)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 842)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 44.569 CPU sec, 45.457 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 41683, 0, 2)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (276, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 702)\n",
            "[M::mem_pestat] mean and std.dev: (348.70, 108.53)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 844)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 56.459 CPU sec, 57.204 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43330, 1, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (278, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 695)\n",
            "[M::mem_pestat] mean and std.dev: (348.63, 106.97)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 834)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 41.529 CPU sec, 41.927 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 36690, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 703)\n",
            "[M::mem_pestat] mean and std.dev: (347.96, 108.15)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 846)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 119.447 CPU sec, 121.899 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41493, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 403, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 703)\n",
            "[M::mem_pestat] mean and std.dev: (348.15, 108.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 846)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 50.501 CPU sec, 53.056 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 41108, 1, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (281, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (9, 689)\n",
            "[M::mem_pestat] mean and std.dev: (349.38, 106.86)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 825)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 44.269 CPU sec, 45.066 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41036, 2, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 704)\n",
            "[M::mem_pestat] mean and std.dev: (348.45, 108.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 847)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 75.950 CPU sec, 77.323 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 43931, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 701)\n",
            "[M::mem_pestat] mean and std.dev: (347.95, 107.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 843)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 45.169 CPU sec, 46.165 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 34180, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (278, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 698)\n",
            "[M::mem_pestat] mean and std.dev: (349.28, 107.85)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 838)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 56.692 CPU sec, 57.646 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 43068, 1, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (277, 401, 416)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 694)\n",
            "[M::mem_pestat] mean and std.dev: (347.34, 107.13)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 833)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 46.045 CPU sec, 46.761 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 29347, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 703)\n",
            "[M::mem_pestat] mean and std.dev: (347.90, 108.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 846)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 91.126 CPU sec, 92.907 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 43231, 0, 2)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 701)\n",
            "[M::mem_pestat] mean and std.dev: (347.81, 107.93)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 843)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 44.995 CPU sec, 45.899 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 43121, 1, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 401, 416)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 700)\n",
            "[M::mem_pestat] mean and std.dev: (347.53, 107.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 842)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 42.360 CPU sec, 43.079 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 39545, 1, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (278, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 695)\n",
            "[M::mem_pestat] mean and std.dev: (348.51, 107.61)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 834)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 67.709 CPU sec, 69.227 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42387, 1, 3)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 703)\n",
            "[M::mem_pestat] mean and std.dev: (347.78, 108.17)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 846)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 47.202 CPU sec, 48.172 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 43345, 1, 2)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 706)\n",
            "[M::mem_pestat] mean and std.dev: (348.39, 108.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 850)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 50.246 CPU sec, 52.539 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 42433, 1, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (276, 401, 416)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 696)\n",
            "[M::mem_pestat] mean and std.dev: (347.64, 107.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 836)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 46.950 CPU sec, 47.789 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 44262, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 404, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 706)\n",
            "[M::mem_pestat] mean and std.dev: (348.75, 108.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 850)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 41.258 CPU sec, 42.171 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 41518, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (277, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 700)\n",
            "[M::mem_pestat] mean and std.dev: (349.27, 107.64)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 841)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 47.760 CPU sec, 48.584 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 38402, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (278, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 695)\n",
            "[M::mem_pestat] mean and std.dev: (348.39, 107.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 834)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 45.724 CPU sec, 46.469 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 42415, 0, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 401, 416)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 698)\n",
            "[M::mem_pestat] mean and std.dev: (347.48, 107.48)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 839)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 45.815 CPU sec, 46.627 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 30366, 2, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (277, 401, 416)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 694)\n",
            "[M::mem_pestat] mean and std.dev: (347.69, 107.68)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 833)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 58.219 CPU sec, 59.340 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 37595, 0, 4)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (272, 400, 415)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 701)\n",
            "[M::mem_pestat] mean and std.dev: (346.25, 107.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 844)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 72.965 CPU sec, 74.322 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 40175, 0, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 401, 416)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 700)\n",
            "[M::mem_pestat] mean and std.dev: (347.13, 107.78)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 842)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 56.785 CPU sec, 57.857 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 43486, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 701)\n",
            "[M::mem_pestat] mean and std.dev: (347.96, 107.64)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 843)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 43.738 CPU sec, 44.936 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 37063, 1, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (276, 400, 415)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 693)\n",
            "[M::mem_pestat] mean and std.dev: (346.84, 106.94)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 832)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 94.735 CPU sec, 97.139 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 38933, 1, 1)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (275, 401, 416)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 698)\n",
            "[M::mem_pestat] mean and std.dev: (347.25, 107.81)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 839)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 52.247 CPU sec, 54.273 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 38345, 1, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (277, 400, 415)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 691)\n",
            "[M::mem_pestat] mean and std.dev: (347.24, 107.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 829)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 48.454 CPU sec, 49.320 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 38169, 0, 4)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (274, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 706)\n",
            "[M::mem_pestat] mean and std.dev: (348.06, 108.84)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 850)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 77.003 CPU sec, 78.061 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 12125, 2, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (276, 402, 417)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 699)\n",
            "[M::mem_pestat] mean and std.dev: (347.51, 108.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 840)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 99010 reads in 208.814 CPU sec, 212.729 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 5783, 225, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (115, 353, 409)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 997)\n",
            "[M::mem_pestat] mean and std.dev: (274.89, 152.70)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1291)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (63, 104, 165)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 369)\n",
            "[M::mem_pestat] mean and std.dev: (108.25, 61.66)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 471)\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_process_seqs] Processed 99010 reads in 282.873 CPU sec, 289.646 real sec\n",
            "[M::process] read 99010 sequences (10000010 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 6414, 240, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (108, 343, 409)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1011)\n",
            "[M::mem_pestat] mean and std.dev: (270.50, 153.96)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1312)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (72, 115, 169)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 363)\n",
            "[M::mem_pestat] mean and std.dev: (114.84, 60.53)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 460)\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_process_seqs] Processed 99010 reads in 312.541 CPU sec, 319.659 real sec\n",
            "[M::process] read 49500 sequences (4999500 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 36332, 21, 3)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (268, 399, 414)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 706)\n",
            "[M::mem_pestat] mean and std.dev: (343.92, 108.81)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 852)\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (64, 101, 159)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 349)\n",
            "[M::mem_pestat] mean and std.dev: (102.95, 50.54)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 444)\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RF\n",
            "[M::mem_process_seqs] Processed 99010 reads in 78.740 CPU sec, 80.224 real sec\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 22023, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (278, 403, 418)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 698)\n",
            "[M::mem_pestat] mean and std.dev: (348.93, 107.51)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 838)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 49500 reads in 21.408 CPU sec, 21.746 real sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa mem Human38/Human HumanWGS.SUBSET.R1.fastq HumanWGS.SUBSET.R2.fastq\n",
            "[main] Real time: 3605.303 sec; CPU: 3496.800 sec\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!head -n 100 Human.sam #show the first 100 lines of the sam file"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IB2mYCYaSnG6",
        "outputId": "d492ca5a-c047-497f-cc28-6243f9d29185"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "@SQ\tSN:chr1\tLN:248956422\n",
            "@SQ\tSN:chr10\tLN:133797422\n",
            "@SQ\tSN:chr11\tLN:135086622\n",
            "@SQ\tSN:chr12\tLN:133275309\n",
            "@SQ\tSN:chr13\tLN:114364328\n",
            "@SQ\tSN:chr14\tLN:107043718\n",
            "@SQ\tSN:chr15\tLN:101991189\n",
            "@SQ\tSN:chr16\tLN:90338345\n",
            "@SQ\tSN:chr17\tLN:83257441\n",
            "@SQ\tSN:chr18\tLN:80373285\n",
            "@SQ\tSN:chr19\tLN:58617616\n",
            "@SQ\tSN:chr2\tLN:242193529\n",
            "@SQ\tSN:chr20\tLN:64444167\n",
            "@SQ\tSN:chr21\tLN:46709983\n",
            "@SQ\tSN:chr22\tLN:50818468\n",
            "@SQ\tSN:chr3\tLN:198295559\n",
            "@SQ\tSN:chr4\tLN:190214555\n",
            "@SQ\tSN:chr5\tLN:181538259\n",
            "@SQ\tSN:chr6\tLN:170805979\n",
            "@SQ\tSN:chr7\tLN:159345973\n",
            "@SQ\tSN:chr8\tLN:145138636\n",
            "@SQ\tSN:chr9\tLN:138394717\n",
            "@SQ\tSN:chrM\tLN:16569\n",
            "@SQ\tSN:chrX\tLN:156040895\n",
            "@SQ\tSN:chrY\tLN:57227415\n",
            "@PG\tID:bwa\tPN:bwa\tVN:0.7.17-r1188\tCL:bwa mem Human38/Human HumanWGS.SUBSET.R1.fastq HumanWGS.SUBSET.R2.fastq\n",
            "SRR588495.1\t83\tchr1\t248945980\t0\t27M1D52M2D11M1D11M\t=\t248945997\t-88\tGGGTTAGGGGTAGGGTTAGGGTTAGGGTAGGGTTAGGGTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTAGGGTTAGGGTAGGGTTAGGG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:5\tMD:Z:9T17^T52^GT11^T11\tMC:Z:72M29S\tAS:i:74\tXS:i:75\n",
            "SRR588495.1\t163\tchr1\t248945997\t0\t72M29S\t=\t248945980\t88\tAGGGTTAGGGTTAGGGTTAGGGTTGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGGGTAGGGTTGGGGGTGGGGGTGGGGGGGGGGGGGGGGG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:2\tMD:Z:23A40T7\tMC:Z:27M1D52M2D11M1D11M\tAS:i:62\tXS:i:63\n",
            "SRR588495.2\t99\tchr1\t10026\t31\t101M\t=\t10275\t292\tAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCAACCCTAACCCTAACCCT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:20S25M2I18M36S\tAS:i:101\tXS:i:94\n",
            "SRR588495.2\t147\tchr1\t10275\t18\t20S25M2I18M36S\t=\t10026\t-292\tCCCCCCCAACCCCACCCCTCACCCTATCCCTAACCCCAACCCCAAACCCCAAACCCCAACCCCAAACCCCAAACCCCAAACCCTAAACCCTAAACCCTAAA\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:4\tMD:Z:6A21C14\tMC:Z:101M\tAS:i:25\tXS:i:28\n",
            "SRR588495.3\t81\tchr10\t133787300\t4\t60M1I40M\tchr2\t113602858\t0\tGGGTGAGGGTGAGGGTGAGGGTGAGGGTGAGGGTGAGGGTTAGGGTGAGGGTGAGGGTGAGGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:6\tMD:Z:10C41T5T11G17G11\tMC:Z:68M33S\tAS:i:68\tXS:i:63\n",
            "SRR588495.3\t161\tchr2\t113602858\t6\t68M33S\tchr10\t133787300\t0\tTTAGGGTTAGGGGTAGGGGTTGGGGTTGGGGTTGGGGTTGGGGGTGGGGTTGGGGTTGGGGTTAGGGTCAGGGTCAGGGTCAGGGTGAGGGTGAGGGGTAG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:4\tMD:Z:2G9T8A21T24\tMC:Z:60M1I40M\tAS:i:50\tXS:i:45\tSA:Z:chr21,46134442,+,61S36M4S,1,0;\tXA:Z:chr1,+248946136,3S60M38S,3;chrY,+57216427,15S53M33S,2;chrX,+156029907,15S53M33S,2;chrX,+156029995,8M1I18M1D41M33S,5;chrY,+57216515,8M1I18M1D41M33S,5;\n",
            "SRR588495.3\t2209\tchr21\t46134442\t1\t61H36M4H\tchr10\t133787300\t0\tTTAGGGTCAGGGTCAGGGTCAGGGTGAGGGTGAGGG\t????????????????????????????????????\tNM:i:0\tMD:Z:36\tMC:Z:60M1I40M\tAS:i:36\tXS:i:34\tSA:Z:chr2,113602858,+,68M33S,6,4;\n",
            "SRR588495.4\t99\tchr22\t50807936\t0\t101M\t=\t50807904\t64\tAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTGAGGGTTCGGGTTAGGGTTAGGGTTAGGGTTAGGGTTCGGGTTAGGGG\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:3\tMD:Z:53T6A39T0\tMC:Z:5S96M\tAS:i:90\tXS:i:90\n",
            "SRR588495.4\t147\tchr22\t50807904\t0\t5S96M\t=\t50807936\t-64\tTAGGATTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGAGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGG\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:1\tMD:Z:46G49\tMC:Z:101M\tAS:i:91\tXS:i:96\n",
            "SRR588495.5\t99\tchr3\t10590\t0\t55M1I39M6S\t=\t10584\t41\tCCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTGACCCTACCACTAACCCTACCCCTCACCCTAACCCTAACCCTAACCCTACACCTA\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:5\tMD:Z:11C34A17A4A24\tMC:Z:54S47M\tAS:i:67\tXS:i:73\n",
            "SRR588495.5\t147\tchr3\t10584\t36\t54S47M\t=\t10590\t-41\tCCCTACCCCTATCTCTACCCATTTCCCAAACCCGAACCCTATCCCTTGCCTTATCCCTACCCCTAGCCCTACCCCTAACCCTAACCCTAACCCTAACCCTA\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:1\tMD:Z:11A35\tMC:Z:55M1I39M6S\tAS:i:42\tXS:i:35\tXA:Z:chr12,-10263,54S47M,3;\n",
            "SRR588495.6\t83\tchr12\t10206\t0\t101M\t=\t10002\t-305\tCAACCCTAACCCTAACCCTACCCCTAACCCTAACCCTAACCCTATCCCTACCCCTATCCCCACCCCTAACCCTACCCCTAACCCTAACCCTAACCCTAACC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:7\tMD:Z:0T19A23A5A5A3T1A38\tMC:Z:101M\tAS:i:70\tXS:i:65\n",
            "SRR588495.6\t163\tchr12\t10002\t0\t101M\t=\t10206\t305\tTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.7\t83\tchr10\t10095\t0\t13S18M1I69M\t=\t10028\t-154\tCCCATAACCATAGCCCTAACCCTAACCCAAACCCCTAACCCTAACCCTAACCCTACCCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:3\tMD:Z:15T25A45\tMC:Z:92M9S\tAS:i:70\tXS:i:70\n",
            "SRR588495.7\t163\tchr10\t10028\t0\t92M9S\t=\t10095\t154\tACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAGCCTAACCCTAACCCTAACCCTAACCCTAACCCTGACCCTAACAACAACACC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:2\tMD:Z:49C33A8\tMC:Z:13S18M1I69M\tAS:i:82\tXS:i:82\n",
            "SRR588495.8\t99\tchr1\t248946026\t3\t35M1I41M24S\t=\t248946064\t89\tAGGGTTAGGGTTAGGGTTAGGGGTAGGGTTAGGGGTTTAGGGTTAGGGTTAGGGTTAGGGATAGGGTTGGGGTTAGGTGTAGGGAAAGGGGCGGGGCAGCG\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:4\tMD:Z:22T36T7A8\tMC:Z:50S51M\tAS:i:54\tXS:i:57\n",
            "SRR588495.8\t147\tchr1\t248946064\t3\t50S51M\t=\t248946026\t-89\tGTACGTCTGAGTGTGGGAGGGGTTCGTGTGGGGGTTGGGTTTGAGGGGTGGGGTTAGGTTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTGGGTTAGGGT\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:1\tMD:Z:8G42\tMC:Z:35M1I41M24S\tAS:i:46\tXS:i:48\n",
            "SRR588495.9\t99\tchr1\t10043\t0\t58M43S\t=\t10396\t398\tTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACAGCTCAGCAGAGCATACGGCTGAGCTGCAGCCACCCCCATACC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:58\tMC:Z:56S45M\tAS:i:58\tXS:i:59\n",
            "SRR588495.9\t147\tchr1\t10396\t0\t56S45M\t=\t10043\t-398\tCACAAGATATCCCCTCCTTCCATACACAACACTCTTCTAAACTTAACCCCACCCCTACCCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:45\tMC:Z:58M43S\tAS:i:45\tXS:i:47\n",
            "SRR588495.10\t83\tchr5\t10590\t0\t101M\t=\t10312\t-379\tACCCCTATCCCTAACCCTAACCCTAGCCCTAAGCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTATCCCTAACCCTAACCCTAACCC\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:5\tMD:Z:1A5A17A6C46A21\tMC:Z:101M\tAS:i:79\tXS:i:84\n",
            "SRR588495.10\t163\tchr5\t10312\t0\t101M\t=\t10590\t379\tCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTATCCCTAACCCTGACCCTAACCCTAACCCTAAC\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:2\tMD:Z:69A10A20\tMC:Z:101M\tAS:i:91\tXS:i:91\n",
            "SRR588495.11\t99\tchr12\t10016\t0\t100M1S\t=\t10178\t243\tACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACACTAACACAAACCCTAACCATAACGCTAACCCTAACACTAGCCCTAACCTA\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:7\tMD:Z:50C5C1T10C4C11C3A9\tMC:Z:19S11M1I70M\tAS:i:65\tXS:i:64\n",
            "SRR588495.11\t147\tchr12\t10178\t0\t19S11M1I70M\t=\t10016\t-243\tAACACATTACCCTTTCACTACCCCTACCCCAAACCCTACCCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACC\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:2\tMD:Z:16A64\tMC:Z:100M1S\tAS:i:69\tXS:i:67\n",
            "SRR588495.12\t83\tchr1\t11765\t0\t101M\t=\t11458\t-408\tGTGTTGAGAATGACTGCGCAAATTTGCCGGATTTCCTTTGCTGTTCCTGCATGTAGTTTAAACGAGATTGCCAGCACCGGGTATCATTCACCATTTTTCTT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:83M1I17M\tAS:i:101\tXS:i:101\n",
            "SRR588495.12\t163\tchr1\t11458\t0\t83M1I17M\t=\t11765\t408\tTGGCGGATTATAGGGAAACACCCGGAGCATATGCTGTTTGGTCTCAGGAGACTCCTAAATATGGGATTCCTGGGTTTAAAAGTAAAAAAATAAATATGTTT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:2\tMD:Z:47T52\tMC:Z:101M\tAS:i:88\tXS:i:88\n",
            "SRR588495.13\t99\tchr1\t182389\t25\t101M\t=\t182677\t389\tTTAACTTGCCGTCAGCCTTTTCTTTGACCTCTTCTTTCTGTTCATGTGTATTTGCTGTCTCTTAGCCCAGACTTCCCGTGTCCTTTCCACCGGGCCTTTGA\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.13\t147\tchr1\t182677\t25\t101M\t=\t182389\t-389\tCAACCAGGCATAGGGGAAAGATTGGAGGAAAGATGAGTGACAGCATCAACTTCTCTCACAACCTAGGCCAGTAAGTAGTGCTTGTGCTCATCTCCTTGGCT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:96\n",
            "SRR588495.14\t99\tchrX\t156027597\t0\t101M\t=\t156027891\t395\tCCACCTAGGCCAATCCACTGGGTTCTGTGGCAGCGATGGCCTGCCTGATCTTCCACCTGCTCTCCCAGGGCCAAAGCCAGACCTGCTGAGCCCCTCCCTCC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.14\t147\tchrX\t156027891\t0\t101M\t=\t156027597\t-395\tGCCTGGTTGTGGTATTAAGTTACATGCAGACAACAGGGGCCAGAAGATGAACAATGGCCCATCCCACTCTAGGCATGGCTCCTCTCCACAGGAAAACTCCA\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.15\t83\tchr15\t101978441\t0\t101M\t=\t101978234\t-308\tCCTGCGGCTCTGCCCCTCCCGTCTCTGCACCTACCACCCAGAGAGGGCTTGTTGAGCTCAGAGATCCCACCTAGGCCAATCCACTGGGTTCTGTGGCAGCG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.15\t163\tchr15\t101978234\t0\t101M\t=\t101978441\t308\tCTCTCCTCTCACCTCTGCAACACTGGGGACACTCACAAGAGTGTGATCCAAGTCGGCCGTCATCTTCTGCAGCTGTGGAGACCTGATGCTGGGGAAGGGCA\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:2\tMD:Z:61G12C26\tMC:Z:101M\tAS:i:91\tXS:i:91\n",
            "SRR588495.16\t99\tchr1\t183667\t0\t101M\t=\t183974\t408\tTGAGGAAGGAGAAGGGGATGCACTGTTGGGAAGGCAGCTGTAACTCAAAGCCTTAGCCTCTGTTCCCACGAAGGCAGGGCCATCAGGCACCAAAGGGATTC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:1\tMD:Z:30G70\tMC:Z:101M\tAS:i:96\tXS:i:96\n",
            "SRR588495.16\t147\tchr1\t183974\t0\t101M\t=\t183667\t-408\tTCACCAGAAGTAGGCCTCTTCCTGACAGGCAGCTGCACCACTGCCTGGCGCTGTGCCCTTCCTTTGCTCTGCCCGCTGGAGACGGTGTTTGTCATGGGCCT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.17\t99\tchr1\t183949\t0\t101M\t=\t184261\t413\tCTCACTGCCTTTTGTCTGCCCAGTTTCACCAGAAGTAGGCCTCTTCCTGACAGGCAGCTGCACCACTGCCTGGCGCTGTGCCCTTCCTTTGCTCTGCCCGC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.17\t147\tchr1\t184261\t0\t101M\t=\t183949\t-413\tATCCTGCGTGGCCGAGGGCCAGGCTTCTCACTGGGCCTCTGCAGGAGGCTGCCATTTGTCCTGCCCACCTTCTTAGAAGCGAGACGGAGCAGACCCATCTG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.18\t99\tchr1\t184533\t0\t101M\t=\t184625\t193\tTGCCTGGATTTCTATCTCCCTGCCTTGGTGCCAGTTCCTCCAAGTCGATGGCACCTCCCTCCCTCTCAACCACTTGAGCAAACTCCAAGACATCTTCTACC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:1\tMD:Z:22G78\tMC:Z:101M\tAS:i:96\tXS:i:96\n",
            "SRR588495.18\t147\tchr1\t184625\t0\t101M\t=\t184533\t-193\tTCTTCTACCCCAACACCAGCAATTGTGCCAAGGGCCATTAGGCTCTCAGCATGACTGTTTTTAGAGACCCCGTGTCTGTCACTGAAACCTTTTTTGTGGGA\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:1\tMD:Z:56A44\tMC:Z:101M\tAS:i:96\tXS:i:96\n",
            "SRR588495.19\t83\tchr16\t14512\t0\t33S68M\t=\t14173\t-407\tCCTCCCCCGCCCCCGGTCCCTTCCCCGGGGTGGCTGGAGGGAAAAGGCTTGGGGGGGGTGGTTGGGGGGAAACCCTGGTTCCCCCAGCCCCCGGAGACTTA\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:5\tMD:Z:16G0A1T1A10T35\tMC:Z:101M\tAS:i:43\tXS:i:43\n",
            "SRR588495.19\t163\tchr16\t14173\t0\t101M\t=\t14512\t407\tGTCCCCCCATGGAGCACAGGCAGACAGAAGTCCCCGCCCCAGCTGTGTGGCCTCAAGCCAGCCTTCCGCTCCTTGAAGCTGGTCTCCACACAGTGCTGGTT\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:0\tMD:Z:101\tMC:Z:33S68M\tAS:i:101\tXS:i:101\n",
            "SRR588495.20\t83\tchr12\t15771\t0\t101M\t=\t15688\t-184\tTTCCAGCATGTGCCCTAGGGGAAGCAGGGGCCAGCTGGCAAGAGCAGGGGGTGGGCAGAAAGCACCCGGTGGACTCAGGGCTGGAGGGGAGGAGGCGATCT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.20\t163\tchr12\t15688\t0\t101M\t=\t15771\t184\tGAAAGCTCCACTATGGCTCCCAAACCAGGAAGGAGCCATAGCCCAGGCAGGAGGGCTGAGGACCTCTGGTGGCGGCCCAGGGCTTCCAGCATGTGCCCTAG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.21\t99\tchr16\t15754\t0\t101M\t=\t16053\t400\tCGCGCTGGGGAGGCCGGACCTTTGGAGACTGTGTGGGGCCCGGGCACTGACTTCGGCAACCACCTGAGCGCGGGCATCCTGTGTGCAGATACTCCCTGCTT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.21\t147\tchr16\t16053\t0\t101M\t=\t15754\t-400\tATTAGGAATGGGGAACCCAGAGCCTCACTTGTTCAGGCTCCCTCTGCCCTAGAAGTGAGAAGTCCAGAGCTCTACAGTTTGAAAACCACTATTTTATGAAC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:2\tMD:Z:10A73G16\tMC:Z:101M\tAS:i:91\tXS:i:101\n",
            "SRR588495.22\t99\tchr1\t187118\t24\t101M\t=\t187428\t411\tGCTGCTCCCACCTGAAGGAGACGCGCTGCTGCTGCTGTCGTCCTGCCTGGCGCCTTGGCCTACAGGGGCCGCGGTTGAGGGTGGGAGTGGGGGTGCACTGG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.22\t147\tchr1\t187428\t24\t101M\t=\t187118\t-411\tGGTGCCAGGGGCAGAGGGGGCAATGCCGGGGCCCAGGTCGGCAATGTACATGAGGTCATTGGCAATGCCGGACAGGTCAGGCAGGTAGGATGGAACATCAA\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:1\tMD:Z:57G43\tMC:Z:101M\tAS:i:96\tXS:i:91\n",
            "SRR588495.23\t99\tchr1\t190368\t0\t101M\t=\t190699\t432\tCAGCACAAGCTGTCTTAATTGACCCTAGTTCCCAGGGCAGCCTCGTTCTGCCTTGGGTGCTGACACGACCTTCGGTAGGTGCATAAGCTCTGCATTCGAGG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\tXA:Z:chr12,+19880,101M,0;chr19,+61453,101M,0;chr1,+19845,101M,0;chr9,+19958,101M,0;chr15,-101971016,101M,2;\n",
            "SRR588495.23\t147\tchr1\t190699\t0\t101M\t=\t190368\t-432\tTTTGTAACAGGATTATTAGCCTGCTGTGCCTGGGGAAAACATGCAGCACAGTGCATCTCGAGTCAGCAGAATTTTGACGGCTTCTAACAAAATCTTGTAGA\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:2\tMD:Z:30C38G31\tMC:Z:101M\tAS:i:91\tXS:i:91\tXA:Z:chr1,-20176,101M,2;chr2,+113593153,101M,2;chr15,+101970685,101M,3;chr9,-20289,101M,4;chr19,-61784,101M,4;\n",
            "SRR588495.24\t83\tchr1\t190928\t23\t101M\t=\t190608\t-421\tTCCATCCAGTTCATTCAACACCTGGTTAGAAAACTGGGGCCAGCACACAGGGGAAGGGTAAGTTGGTTTCATGATCGAATCAAGGCTCAGACAATTTTTAA\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:96\n",
            "SRR588495.24\t163\tchr1\t190608\t23\t101M\t=\t190928\t421\tATGGATTATTAAACAGAGAGTCTGTAAGCACTTAGAAAAGGCCGCGGTGAGTCCCAGGGGCCAGCACTGCTCGAAATGTACAGCATTTCTCTTTGTAACAG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.25\t83\tchr1\t191803\t28\t101M\t=\t191696\t-208\tCGTCGTTTCCTCTGAATGTCTCACCCTGCCCTCCCTGCTTGCAAGTCCCCTGTCTGCAGCCTCACCCCTGTCGCATCCTGACTACAAAAACAGCTTCTGGG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:1\tMD:Z:87T13\tMC:Z:101M\tAS:i:96\tXS:i:91\n",
            "SRR588495.25\t163\tchr1\t191696\t28\t101M\t=\t191803\t208\tAGGGGCAAGTGGAGGAGGAGAGGTGGCGGTGCTCCCCACTCCACTGCCAGTCGTCACTGGCTCTCCCTTCCCTTCATCCTCGTTCCCTATCTGTCACCATT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.26\t83\tchr9\t26040\t0\t101M\t=\t25712\t-429\tTCCTTTCGATTTCCCATTCTCCAAACTGAGAACTAGCACTGCTGAGACGTGGTTATTTCCAATAATAATTTGTATATTTTACATAACGCACCACACCAACA\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\tXA:Z:chr19,-67535,101M,0;chr1,-25927,101M,0;chr15,+101964933,101M,1;chr12,-27990,87M14S,1;chr1,-196452,87M14S,1;\n",
            "SRR588495.26\t163\tchr9\t25712\t0\t101M\t=\t26040\t429\tTAATTAACTCCTCTCCCTGGTGCCACTAAAGCAGCAATCCCACTGCAGACAGCACTGATTTGATTGGCAAGAGATGCACCAGGCAGAATATTAAGGGACGA\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:2\tMD:Z:39A59C1\tMC:Z:101M\tAS:i:94\tXS:i:94\n",
            "SRR588495.27\t83\tchr19\t67782\t0\t101M\t=\t67454\t-429\tGAGCCCAGGAGTTACTCCTCACCTCCCACTTATTTTTCCTCCCATCAAATAACTAAAGCATGGCCAGCTGATGCCCAGCCAACTGAGAAACCCAACCCTCT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.27\t163\tchr19\t67454\t0\t101M\t=\t67782\t429\tAGTGTGGCATGAAGGGCCTAGGAGATTTCACTTGGGGTTAAAATGCTGTGACCTTGAGTAAGTTGCCGTCTCTGAATCTGATCCTTTCGGTTTCCCATTCT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:2\tMD:Z:36T52A11\tMC:Z:101M\tAS:i:91\tXS:i:91\n",
            "SRR588495.28\t83\tchr19\t68049\t0\t101M\t=\t67747\t-403\tGAATTCCTGGTTAGTGTGTGTTTGTTTGCTCACACATATGCGTGAAAGAAGAGTGCATACCCCAATCCCCTCAAAAAGCAGGCAGCAAGCCCGTTCAAGAA\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:9\tMD:Z:5T2T12G32A5A1A0G3T10G22\tMC:Z:101M\tAS:i:57\tXS:i:57\n",
            "SRR588495.28\t163\tchr19\t67747\t0\t101M\t=\t68049\t403\tCTTTCCAGATCTGTACTTGCCTTGAAAACTCATCAGATCCCACGAGTTACTCCTCACCTCCCAATCATTTTTACTCCCACCAAATCACTCAAGCATGACCA\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:10\tMD:Z:26T10G4G20C1T6C6T5A3A7G3\tMC:Z:101M\tAS:i:52\tXS:i:52\n",
            "SRR588495.29\t83\tchr1\t197269\t0\t101M\t=\t196940\t-430\tTTAGGGGAGTCAGGAAAAAGCTCTTAGCAGAATGTGTGCCTCTCGGCCGGGCGCAGCGGCTCACGCCTGTAATCCCAGCACTTTGGGAGGCGAAGGCAGGC\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.29\t163\tchr1\t196940\t0\t101M\t=\t197269\t430\tATTTCCTGTGCACATCTTCAGATGAATTTCTTGTTAGTGTGTGTGTGTTTGCTCACACATATGCGTGAAAGAAGAGTACATACACAGATCTCCTCAAAAAG\t$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.30\t83\tchr1\t26747\t0\t101M\t=\t26418\t-430\tTTAGGGGAGTCAGGAAAAAGCTCTTAGCAGAATGTGTGCCTCTCGGCCGGGCGCAGCGGCTCACGCCTGTAATCCCAGCACTTTGGGAGGCGAAGGCAGGC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.30\t163\tchr1\t26418\t0\t101M\t=\t26747\t430\tATTTCCTGAGCACATCTTCAGATGAATTTCTTGTTAGTGTGTGTGTGTTTGCTCACACATATGCGTGAAAGAAGAGTACATACACAGATCTCCTCAAAAAG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:1\tMD:Z:8T92\tMC:Z:101M\tAS:i:96\tXS:i:96\n",
            "SRR588495.31\t99\tchr9\t30093\t0\t101M\t=\t30702\t705\tCCCTCCGGCATTTTTGGCCCTTGCCTTTTAGGGTTGCCAGATTAAAAGACAGGATGCCCAGTTAGTTTGAATTTTAGATAAACAACGAATAATTTCTTAGC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:3\tMD:Z:0G60C34G4\tMC:Z:7S36M2D58M\tAS:i:90\tXS:i:90\n",
            "SRR588495.31\t147\tchr9\t30702\t0\t7S36M2D58M\t=\t30093\t-705\tTTCTCCGCTCTCTCTCTCATTTCGCTCTCTCTCTCTTTCTCTCCTGTCTTTTCCCACCAAGTGAGGATGCGAAGAGAAGGTGACTGTCTGCAAACCAGGAA\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:6\tMD:Z:6A8T0T19^CT39G18\tMC:Z:101M\tAS:i:66\tXS:i:66\n",
            "SRR588495.32\t83\tchr19\t75667\t0\t101M\t=\t75348\t-420\tTTTAAAGCAGTGTGTAAAGAGACATTTATAGCACTAAATGCCCACAAGAGACCTCTGCCTGAGAACGTGGGTTTCAGCCTAAGAGTTGTAATATGTGTGCC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.32\t163\tchr19\t75348\t0\t101M\t=\t75667\t420\tATTGTTCAATTCCCACCTGTGAGTGAGAACATGCAGTGTTTGGTTTTCTGTCTTTGTGATAGTTTGCTCAGAATGATGGTTTCCAGCTTCATCCATGTCCC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.33\t99\tchr2\t113577401\t0\t101M\t=\t113577448\t148\tGCTGTGGACCCTGTGCCAGGAAAGGAAGGGCGCAGCTCCTGCAATGCGGAGCAGCCAGGGCAGTGGGCACCAGGCTTTAGCCTCCCTTTCTCACCCTACAG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.33\t147\tchr2\t113577448\t0\t101M\t=\t113577401\t-148\tGGAGCAGCCAGGGCAGTGGGCACCAGGCTTTAGCCTCCCTTTCTCACCCTACAGAGGGCAGGCCCTTCAGCTCCATTCTCCTCCAAGGCTGCAGAGGGGGC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.34\t83\tchr19\t78268\t0\t101M\t=\t77956\t-413\tGAGAAAGCCTCCTGATCCTGAAAGCAAGAATTTGACCAGAGCAGAAGTAATCAGTATGCAGATTGATTCTGTGGTATGTTAATGTTTATGCATAGATTATG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.34\t163\tchr19\t77956\t0\t101M\t=\t78268\t413\tTTTGGAAAAAAAAAAAAACCCAGCCTGGCGGAAAGAATTTAAATTATAAAAACTTAGAAGTATGGAATGTGAAATCATCCTGTAGGTGCTTATTTAACAAC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.35\t83\tchr9\t36516\t0\t101M\t=\t36357\t-260\tTATGTTAACGTTTATGCATAGATTATGAGGACCAGGTGAAAAGTGGGCCAGGGGAGCCAGATGTGTGTGTGAGTCATGGGTGGCTGAGATGAGGACAGGAG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:1\tMD:Z:8T92\tMC:Z:101M\tAS:i:96\tXS:i:96\tXA:Z:chr1,-36733,101M,1;chr19,-78342,101M,1;chr15,+101954132,101M,1;chr2,+113576630,101M,3;\n",
            "SRR588495.35\t163\tchr9\t36357\t0\t101M\t=\t36516\t260\tCACCCCTTGTCTCATTAGACATGCAACTACGGGAGGGGTGACAGGAAGACAAGACACTATTTCCTCAGGCCCAGTTTGGTGTGGGGAGAAAGCCTCCTGAT\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\n",
            "SRR588495.36\t83\tchr19\t79505\t0\t101M\t=\t79419\t-187\tCCAGAGACCTTAAGGAGGGGAGACCAGGTCCTGAGTAAAGTTGAAGGGGAGGGGCTGAGTCCTGCTAGCCAGGAGTCTCATCCCCTGGGGAAGTTCCAGGG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\tXA:Z:chr12,-39947,101M,0;chr1,-37896,101M,0;chr9,-37678,101M,0;chr15,+101952970,101M,0;chr2,+113575457,101M,1;\n",
            "SRR588495.36\t163\tchr19\t79419\t0\t101M\t=\t79505\t187\tCATGCTTCACGCGCAGAAGACTCTTCGTTTATATTTGGTGGACATAGAGAAGGCCTGTGATAGTGGGCTGAAGTGTGCGTTGGGCTCCAGAGACCTTAATG\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:9\tMD:Z:6G3T1A16G6T7G6T8G38G1\tMC:Z:101M\tAS:i:61\tXS:i:61\tXA:Z:chr9,+37592,101M,9;chr15,-101953056,101M,9;chr1,+37810,101M,9;chr2,-113575543,101M,10;chr12,+39874,13S88M,7;\n",
            "SRR588495.37\t83\tchr15\t101953260\t0\t101M\t=\t101952940\t-421\tGCACAGGAACAAAGACGGCAGGGGCACAGGGCAATTCAGGGGTGCAGATCTGGGGGCATCCAAGAGGCTCCTAAGCCTCAGTATATAAACTGACAGGTCAC\t?????????????????????????????????????????????????????????????????????????????????????????????????????\tNM:i:0\tMD:Z:101\tMC:Z:101M\tAS:i:101\tXS:i:101\tXA:Z:chr19,+79215,101M,0;chr9,+37388,101M,0;chr2,-113575747,101M,0;chr1,+37606,101M,0;chr12,+39657,101M,2;\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp HumanWGS.SUBSET.R1.fastq drive/MyDrive/GenomicDataVisualization/\n",
        "\n",
        "#this is copying the downloaded file to the google drive,\n",
        "#specifically in the GenomicDataVisualiziation folder"
      ],
      "metadata": {
        "id": "JUNO9ePMgbSv"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp HumanWGS.SUBSET.R2.fastq drive/MyDrive/GenomicDataVisualization/\n",
        "\n",
        "#copying the downloaded files into google drive GenomicDataViz folder"
      ],
      "metadata": {
        "id": "YmHjGSA-gsp4"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp Human.sam drive/MyDrive/GenomicDataVisualization/\n",
        "\n",
        "#copying the sam file into google drive GenomicDataViz folder"
      ],
      "metadata": {
        "id": "i0xAxBG3g1zO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# !bwa aln Human38/Human drive/MyDrive/GenomicDataVisualization/HumanWGS.SUBSET.R1.fastq > drive/MyDrive/GenomicDataVisualization/Test.sai\n",
        "\n",
        "#this is calling the file from google drive\n",
        "#if you delete the code after the > it'll store the file in your environment (I think lol)"
      ],
      "metadata": {
        "id": "-wvl1ZPRhR5P"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Start here:"
      ],
      "metadata": {
        "id": "Q5r5um1sCEGX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!bwa aln Human38/Human drive/MyDrive/GenomicDataVisualization/Human.sam > Humans.sam\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RtZQ2I-hCDuM",
        "outputId": "a9c5f701-fbc9-44d8-ab0b-3bc611af68f4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bwa_aln] 17bp reads: max_diff = 2\n",
            "[bwa_aln] 38bp reads: max_diff = 3\n",
            "[bwa_aln] 64bp reads: max_diff = 4\n",
            "[bwa_aln] 93bp reads: max_diff = 5\n",
            "[bwa_aln] 124bp reads: max_diff = 6\n",
            "[bwa_aln] 157bp reads: max_diff = 7\n",
            "[bwa_aln] 190bp reads: max_diff = 8\n",
            "[bwa_aln] 225bp reads: max_diff = 9\n",
            "[bwa_aln_core] calculate SA coordinate... 0.06 sec\n",
            "[bwa_aln_core] write to the disk... 0.00 sec\n",
            "[bwa_aln_core] 1 sequences have been processed.\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa aln Human38/Human drive/MyDrive/GenomicDataVisualization/Human.sam\n",
            "[main] Real time: 32.174 sec; CPU: 8.489 sec\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "ref_file = \"/content/drive/MyDrive/GenomicDataVisualization/Human.sam\"\n",
        "ref_file"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 35
        },
        "id": "GSTlBEtKFmAr",
        "outputId": "adaa5bc2-eefc-4163-9dfe-4a9cd2317202"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'/content/drive/MyDrive/GenomicDataVisualization/Human.sam'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 29
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools stats ref_file"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Qj0dR6u9jF29",
        "outputId": "0977ddd9-45ae-4406-fe52-ce0ba9810670"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[E::hts_open_format] Failed to open file \"ref_file\" : No such file or directory\n",
            "samtools stats: failed to open \"ref_file\": No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!grep 'chr' Humans.sam | wc\n",
        "\n",
        "#4981439 reads mapped to the reference genome"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RWLlMB4_jZxf",
        "outputId": "67273409-6c3a-46f1-8d44-a4e1b6ec3c94"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "      0       0       0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Mount Google Drive\n",
        "from google.colab import drive\n",
        "drive.mount('/content/drive')\n",
        "\n",
        "# Load the SAM file\n",
        "sam_file_path = \"/content/drive/MyDrive/GenomicDataVisualization/Human.sam\"\n",
        "\n",
        "# Open the SAM file and read its contents\n",
        "with open(sam_file_path, 'r') as sam_file:\n",
        "    sam_contents = sam_file.read()\n",
        "\n",
        "# Print the first few lines of the SAM file\n",
        "print(sam_contents[:1000])  # Print first 1000 characters as an example\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "G6eiNiXLI0ZS",
        "outputId": "9f8100fb-6606-4c59-ecf8-cd09d155c031"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n",
            "@SQ\tSN:chr1\tLN:248956422\n",
            "@SQ\tSN:chr10\tLN:133797422\n",
            "@SQ\tSN:chr11\tLN:135086622\n",
            "@SQ\tSN:chr12\tLN:133275309\n",
            "@SQ\tSN:chr13\tLN:114364328\n",
            "@SQ\tSN:chr14\tLN:107043718\n",
            "@SQ\tSN:chr15\tLN:101991189\n",
            "@SQ\tSN:chr16\tLN:90338345\n",
            "@SQ\tSN:chr17\tLN:83257441\n",
            "@SQ\tSN:chr18\tLN:80373285\n",
            "@SQ\tSN:chr19\tLN:58617616\n",
            "@SQ\tSN:chr2\tLN:242193529\n",
            "@SQ\tSN:chr20\tLN:64444167\n",
            "@SQ\tSN:chr21\tLN:46709983\n",
            "@SQ\tSN:chr22\tLN:50818468\n",
            "@SQ\tSN:chr3\tLN:198295559\n",
            "@SQ\tSN:chr4\tLN:190214555\n",
            "@SQ\tSN:chr5\tLN:181538259\n",
            "@SQ\tSN:chr6\tLN:170805979\n",
            "@SQ\tSN:chr7\tLN:159345973\n",
            "@SQ\tSN:chr8\tLN:145138636\n",
            "@SQ\tSN:chr9\tLN:138394717\n",
            "@SQ\tSN:chrM\tLN:16569\n",
            "@SQ\tSN:chrX\tLN:156040895\n",
            "@SQ\tSN:chrY\tLN:57227415\n",
            "@PG\tID:bwa\tPN:bwa\tVN:0.7.17-r1188\tCL:bwa mem Human38/Human HumanWGS.SUBSET.R1.fastq HumanWGS.SUBSET.R2.fastq\n",
            "SRR588495.1\t83\tchr1\t248945980\t0\t27M1D52M2D11M1D11M\t=\t248945997\t-88\tGGGTTAGGGGTAGGGTTAGGGTTAGGGTAGGGTTAGGGTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTAGGGTTAGGGTAGGGTTAGGG\t????????????????????????????????????????????????????????????????????????????????????????????????\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "sam_file_path = \"/content/drive/MyDrive/GenomicDataVisualization/Human.sam\"\n",
        "\n",
        "# Use samtools to convert the SAM file to BAM, filter by quality, and sort the BAM file\n",
        "!samtools view -bS -q 12 {sam_file_path} | samtools sort -o {sam_file_path}.sorted"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ganLB-QNIAT1",
        "outputId": "48b1dae8-bb14-4c47-9880-a79b14863029"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view {sam_file_path}.sorted\n",
        "\n",
        "#this worked, just takes a long time to run"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "B_Z7XJX4LFWM",
        "outputId": "c6fb1a6e-5463-45d5-8a5f-bb47930711ce"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[main] unrecognized command 'head'\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -bS -q 12 {sam_file_path} | samtools sort -o {sam_file_path}.sorted\n",
        "!bcftools mpileup -f Human38/Human.fasta --max-depth  2000 {sam_file_path}.sorted  | bcftools call --multiallelic-caller --variants-only --ploidy 1  -mv -Oz -o Human.Variants.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "r8GNVUc-LWyW",
        "outputId": "a5044c66-b4fb-4b5e-8e41-50bd4b9e84d5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
            "[mpileup] 1 samples in 1 input files\n",
            "[mpileup] maximum number of reads per input file set to -d 2000\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -df Human.Variants.vcf.gz"
      ],
      "metadata": {
        "id": "C133RaXDZX42"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "####### This is to get you guys excited about next class/(es)\n",
        "!pip install cyvcf2\n",
        "!gzip -df Human.Variants.vcf.gz\n",
        "vcf_path = 'Human.Variants.vcf'"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8-wJxDcfZjEb",
        "outputId": "15a06a8b-3c52-4cc7-e5d3-bb08224f38f9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting cyvcf2\n",
            "  Downloading cyvcf2-0.30.28-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (6.8 MB)\n",
            "\u001b[2K     \u001b[90m\u001b[0m \u001b[32m6.8/6.8 MB\u001b[0m \u001b[31m13.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from cyvcf2) (1.25.2)\n",
            "Collecting coloredlogs (from cyvcf2)\n",
            "  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n",
            "\u001b[2K     \u001b[90m\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m4.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: click in /usr/local/lib/python3.10/dist-packages (from cyvcf2) (8.1.7)\n",
            "Collecting humanfriendly>=9.1 (from coloredlogs->cyvcf2)\n",
            "  Downloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
            "\u001b[2K     \u001b[90m\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m8.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: humanfriendly, coloredlogs, cyvcf2\n",
            "Successfully installed coloredlogs-15.0.1 cyvcf2-0.30.28 humanfriendly-10.0\n",
            "gzip: Human.Variants.vcf.gz: No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cat Human.Variants.vcf"
      ],
      "metadata": {
        "id": "_GkRIbd0Z4Is"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import cyvcf2\n",
        "\n",
        "# Create a VCF reader object\n",
        "vcf_reader = cyvcf2.VCF(vcf_path)\n",
        "\n",
        "with open('output.tsv', 'w') as file:\n",
        "# Loop through each variant in the VCF file\n",
        "    for variant in vcf_reader:\n",
        "    # Access various properties of the variant\n",
        "    ### Se note below\n",
        "        genotype = variant.gt_types[0]\n",
        "        data_string = f\"{variant.CHROM}\\t{variant.POS}\\t{variant.ID}\\t{variant.REF}\\t{','.join(variant.ALT)}\\t{variant.QUAL}\\t{variant.FILTER}\\t{genotype}\\n\"\n",
        "        # Write the data string to the file\n",
        "        file.write(data_string)"
      ],
      "metadata": {
        "id": "BVUU5HUDaGt_"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!more output.tsv"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "21Apn1DEaQ2u",
        "outputId": "64255c2b-29d1-4a5f-efb6-0ec90df55114"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr1\t54716\tNone\tC\tT\t3.2245099544525146\tNone\t3\n",
            "chr1\t54809\tNone\tT\tC\t3.2245099544525146\tNone\t3\n",
            "chr1\t54812\tNone\tT\tC\t3.2245099544525146\tNone\t3\n",
            "chr1\t56279\tNone\tT\tC\t3.2245099544525146\tNone\t3\n",
            "chr1\t57049\tNone\tT\tA\t3.2245099544525146\tNone\t3\n",
            "chr1\t57050\tNone\tC\tA\t3.2245099544525146\tNone\t3\n",
            "chr1\t58195\tNone\tT\tG\t3.2245099544525146\tNone\t3\n",
            "chr1\t58771\tNone\tT\tC\t3.2245099544525146\tNone\t3\n",
            "chr1\t67126\tNone\tA\tC\t3.2245099544525146\tNone\t3\n",
            "chr1\t82734\tNone\tT\tC\t3.2245099544525146\tNone\t3\n",
            "chr1\t83933\tNone\tA\tG\t3.2245099544525146\tNone\t3\n",
            "chr1\t191944\tNone\tG\tA\t20.453500747680664\tNone\t3\n",
            "chr1\t264750\tNone\tT\tG\t3.2245099544525146\tNone\t3\n",
            "chr1\t283378\tNone\tG\tC\t6.512479782104492\tNone\t3\n",
            "chr1\t287134\tNone\tT\tA\t3.2245099544525146\tNone\t3\n",
            "chr1\t287147\tNone\tG\tT\t3.2245099544525146\tNone\t3\n",
            "chr1\t287150\tNone\tC\tA\t3.2245099544525146\tNone\t3\n",
            "chr1\t287157\tNone\tA\tC\t3.2245099544525146\tNone\t3\n",
            "chr1\t287160\tNone\tA\tT\t3.2245099544525146\tNone\t3\n",
            "chr1\t287171\tNone\tT\tA\t3.2245099544525146\tNone\t3\n",
            "chr1\t287173\tNone\tC\tA\t3.2245099544525146\tNone\t3\n",
            "chr1\t287176\tNone\tC\tA\t3.2245099544525146\tNone\t3\n",
            "chr1\t287177\tNone\tC\tA\t3.2245099544525146\tNone\t3\n",
            "\u001b[K"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas\n",
        "\n",
        "\n",
        "columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'GENOTYPE']\n",
        "df = pandas.read_csv(\"output.tsv\", sep = \"\\t\", header=None, names= columns)"
      ],
      "metadata": {
        "id": "QQ0HqShGaUk0"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "df"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 424
        },
        "id": "R_FVXyT7aXvP",
        "outputId": "3479572e-63df-4026-cc1d-4a6c710f5625"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "        CHROM       POS  ID          REF            ALT      QUAL  FILTER  \\\n",
              "0        chr1     54716 NaN            C              T   3.22451     NaN   \n",
              "1        chr1     54809 NaN            T              C   3.22451     NaN   \n",
              "2        chr1     54812 NaN            T              C   3.22451     NaN   \n",
              "3        chr1     56279 NaN            T              C   3.22451     NaN   \n",
              "4        chr1     57049 NaN            T              A   3.22451     NaN   \n",
              "...       ...       ...  ..          ...            ...       ...     ...   \n",
              "1353651  chrY  56882959 NaN            T              C   3.22451     NaN   \n",
              "1353652  chrY  56882990 NaN            G              C   3.22451     NaN   \n",
              "1353653  chrY  56885560 NaN            A              T   3.22451     NaN   \n",
              "1353654  chrY  56885633 NaN  taaaaaaaaaa  tAAaaaaaaaaaa  26.42840     NaN   \n",
              "1353655  chrY  56885653 NaN            G              T   3.22451     NaN   \n",
              "\n",
              "         GENOTYPE  \n",
              "0               3  \n",
              "1               3  \n",
              "2               3  \n",
              "3               3  \n",
              "4               3  \n",
              "...           ...  \n",
              "1353651         3  \n",
              "1353652         3  \n",
              "1353653         3  \n",
              "1353654         3  \n",
              "1353655         3  \n",
              "\n",
              "[1353656 rows x 8 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-8b9ebf5d-c276-43d2-a925-6a69ba2c7cf9\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>ID</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>QUAL</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>GENOTYPE</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>54716</td>\n",
              "      <td>NaN</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>3.22451</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr1</td>\n",
              "      <td>54809</td>\n",
              "      <td>NaN</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>3.22451</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr1</td>\n",
              "      <td>54812</td>\n",
              "      <td>NaN</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>3.22451</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr1</td>\n",
              "      <td>56279</td>\n",
              "      <td>NaN</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>3.22451</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr1</td>\n",
              "      <td>57049</td>\n",
              "      <td>NaN</td>\n",
              "      <td>T</td>\n",
              "      <td>A</td>\n",
              "      <td>3.22451</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1353651</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56882959</td>\n",
              "      <td>NaN</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>3.22451</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1353652</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56882990</td>\n",
              "      <td>NaN</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>3.22451</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1353653</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56885560</td>\n",
              "      <td>NaN</td>\n",
              "      <td>A</td>\n",
              "      <td>T</td>\n",
              "      <td>3.22451</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1353654</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56885633</td>\n",
              "      <td>NaN</td>\n",
              "      <td>taaaaaaaaaa</td>\n",
              "      <td>tAAaaaaaaaaaa</td>\n",
              "      <td>26.42840</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1353655</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56885653</td>\n",
              "      <td>NaN</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>3.22451</td>\n",
              "      <td>NaN</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1353656 rows  8 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-8b9ebf5d-c276-43d2-a925-6a69ba2c7cf9')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-8b9ebf5d-c276-43d2-a925-6a69ba2c7cf9 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-8b9ebf5d-c276-43d2-a925-6a69ba2c7cf9');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-cedaf005-7c67-4431-8e78-ece075bab71d\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-cedaf005-7c67-4431-8e78-ece075bab71d')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-cedaf005-7c67-4431-8e78-ece075bab71d button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df"
            }
          },
          "metadata": {},
          "execution_count": 55
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp Human.Variants.vcf drive/MyDrive/GenomicDataVisualization/\n",
        "#copying the vcf file to my google drive"
      ],
      "metadata": {
        "id": "f2hI-K32alFY"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp output.tsv drive/MyDrive/GenomicDataVisualization/\n"
      ],
      "metadata": {
        "id": "1jIpgnyHcRcC"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}